bakery_cli/fixers.py

```diff
@@ -83,23 +83,68 @@ class VersionFixer(Fixer):
     def get_shell_command(self):
         return 'fontbakery-fix-version.py {}'.format(self.fontpath)
 
-    def increment_major(self):
-        self.font.tables["CFF "].cff.major += 1
-    
-    def increment_minor(self):
-        self.font.tables["CFF "].cff.minor += 1
-
-    def apply(self):
+    def parse_version_string(self, s):
+        try:
+            suffix = ''
+            if ';' in s:
+                fields = s.split(';')
+                s = fields[0]
+                fields.pop(0)
+                suffix = ';'.join(fields)
+
+            substrings = s.split('.')
+            minor = substrings[-1]
+            if ' ' in substrings[-2]:
+                major = int(substrings[-2].split(' ')[-1])
+            else:
+                major = int(substrings[-2])
+            return major, minor, suffix
+        except:
+            print ("ER: failed to detect major and minor version numbers in '{}' utf8 encoding: {}".format(s, [s.encode('utf8')]))
+
+    def increment_version_string(self, s, major_inc, minor_inc, head_table=False):
+        v = self.parse_version_string(s)
+        if v is None:
+            return s
+
+        major = int(v[0]) + major_inc
+        minor = int(v[1]) + minor_inc
```

Comment:
This is an integer value representing the major version (then I add major_inc, the requested major increment).
Same thing for the minor version integer value.